pkg/argocd/update.go

```diff
@@ -516,6 +523,57 @@ func mergeKustomizeOverride(t *kustomizeOverride, o *kustomizeOverride) {
 	}
 }
 
+// Check if a key exists in a MapSlice and return its index and value
+func findHelmValuesKey(m yaml.MapSlice, key string) (int, bool) {
+	for i, item := range m {
+		if item.Key == key {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
+// set value of the parameter passed from the annotations.
+func setHelmValue(m yaml.MapSlice, key string, value interface{}) error {
+	// Check if the full key exists
+	if idx, found := findHelmValuesKey(m, key); found {
+		m[idx].Value = value
+		return nil
+	}
+
+	keys := strings.Split(key, ".")
+	current := m
+
+	for i, k := range keys {
+		found := false
+		for j, item := range current {
+			if item.Key == k {
+				if i == len(keys)-1 {
+					// If we're at the final key, set the value
+					current[j].Value = value
+					found = true
+					break
```

Comment:
Can we just return here since we've set the value and nothing else needs to be done?